opentelemetry-otlp/src/lib.rs

```diff
@@ -396,7 +406,53 @@ pub enum Protocol {
     HttpJson,
 }
 
+impl Display for Protocol {
```

Comment:
What's the motivation behind adding a `Display` implementation for `Protocol`?

Reply:
After seeing the `Compression` implementation within this library had both `FromStr` and `Display`, I felt it was most correct to add `Display` to match the `FromStr` here too.

If you have concerns, I am happy to remove it - the `FromStr` is the functionality I need at present (as per above comment).